commit/plugin_e2e_test.go

```diff
@@ -0,0 +1,343 @@
+package commit
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"sort"
+	"testing"
+	"time"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	libocrtypes "github.com/smartcontractkit/libocr/ragep2p/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-ccip/chainconfig"
+	"github.com/smartcontractkit/chainlink-ccip/internal/libs/testhelpers"
+	"github.com/smartcontractkit/chainlink-ccip/internal/mocks"
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+	reader_mock "github.com/smartcontractkit/chainlink-ccip/mocks/internal_/reader"
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+	"github.com/smartcontractkit/chainlink-ccip/plugintypes"
+)
+
+const (
+	destChain    = ccipocr3.ChainSelector(1)
+	sourceChain1 = ccipocr3.ChainSelector(2)
+	sourceChain2 = ccipocr3.ChainSelector(3)
+)
+
+func TestPlugin_E2E_AllNodesAgree(t *testing.T) {
+	require.Len(t, mapset.NewSet(destChain, sourceChain1, sourceChain2).ToSlice(), 3)
+	require.Less(t, sourceChain1, sourceChain2, "test below expected source chain1 to be LT chain2")
```

Comment:
These seem unnecessary.

Reply:
i have it just in case someone updates the test